puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/pulpcore_packaging_nightly.groovy

```diff
@@ -0,0 +1,69 @@
+def packaging_branch = 'rpm/3.7'
+def source_server = 'http://ah.testing.ansible.com/nightlies/latest/'
+
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 4, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Clone Packaging') {
+            steps {
+                checkout([
+                    $class : 'GitSCM',
+                    branches : [[name: "*/${packaging_branch}"]],
+                    extensions: [[$class: 'CleanCheckout']],
+                    userRemoteConfigs: [
+                        [url: 'https://github.com/theforeman/pulpcore-packaging']
+                    ]
+                ])
+
+            }
+        }
+
+        stage('Nightly Build Packages') {
+            steps {
+                setup_obal()
+
+                script{
+
+                    def response = sh returnStdout: true, script: "curl --silent ${source_server}/index.json"
+                    def index = readJSON text: response
```

Comment:
Have you thought about error handling?

Reply:
Not so far. Do you mean handling bad JSON or HTTP errors?

Reply:
Yes